### PR TITLE
Add Support for Compatible Hive Views

### PR DESCRIPTION
**Pull request by amoghmargoor**
Presto currently doesnot support Hive Views. This change would add support for those Hive Views which is described using HiveQL compatible (or parseable) with Presto.
Following will be the behaviour of compatible and incompatible Hive Views:
1. Show tables will show both compatible and Incompatible Views.
2. Executing against compatible view will succeed provided there are right authorization, where as Executing against incompatible view will fail with Table not Found exception.
3. Authorization is supported.
4. Queries on Information schema will not list columns for incompatible view i.e., queries like `show columns for` will give empty results. 

There are test cases added to capture above behavior

### CODE REVIEW

**Comment by martint** [comment]
This doesn't seem like the right approach to supporting Hive views. HQL and SQL have different semantics, in general (type coercion rules, function behaviors, null handling). Doing a syntactic translation and hoping the query runs can result in different answers between the two engines.

**Comment by amoghmargoor** [comment]
Hi @martint,
It's a very valid observation and something that I had considered too. I do agree if these Hive views were materialized views. For e.g., if Presto was rewriting a query to use these materialized views described by HiveQL query as an optimization then that would be wrong,  as materialized data may not correspond to same query in Presto. But here these views are just logical constructs and will get expanded against actual Hive tables before execution. So just like a query against Hive table can produce different results in different engines, it is acceptable for a queries against virtual view to produce different results as queries against virtual views will get expanded to queries against actual Hive tables.

**Comment by OElesin** [comment]
Hi @amoghmargoor, were you able to achieve this compatibility?

@martint, how you suggest this be done?

**Comment by amoghmargoor** [comment]
Hi @OElesin, As these are just Virtual Views, I am just checking if the views are parseable for compatibility.

**Comment by OElesin** [comment]
Oh @amoghmargoor, looking through your commits to understand what's going on. Were you able to verify the compatibility?

**Comment by amoghmargoor** [comment]
@OElesin Its checking for it here: https://github.com/prestodb/presto/pull/9031/files#diff-dfecb0b24d9d5c723d964cb59873b44eR956

**Comment by OElesin** [comment]
@amoghmargoor, excellent. Were you able to `mvn package`? Currently experiencing issues doing that. I wan to test the implementation

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveUtil.java [hunk 1]
```diff
@@ -514,6 +515,11 @@ public static boolean isPrestoView(Table table)
         return "true".equals(table.getParameters().get(PRESTO_VIEW_FLAG));
     }
 
+    public static boolean isHiveView(Table table)
+    {
+        return table.getTableType().equals(TableType.VIRTUAL_VIEW.name());
```
Comment (jzhuge) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveUtil.java]: For Presto view, the table type is also VIRTUAL_VIEW.

Comment (amoghmargoor) [reply presto-hive/src/main/java/com/facebook/presto/hive/HiveUtil.java]: @jzhuge sorry missed checking this message ... are you interested in this patch ? I can rebase and fix this if you require it.

Comment (jzhuge) [reply presto-hive/src/main/java/com/facebook/presto/hive/HiveUtil.java]: That'd great. Is there a JIRA number for this PR?

More comments to follow.

Comment (amoghmargoor) [reply presto-hive/src/main/java/com/facebook/presto/hive/HiveUtil.java]: @jzhuge This is the issue: https://github.com/prestodb/presto/issues/7338 ... I will try to rebase it and fix your comment.

**Comment by billywatson** [comment]
quick bump: what all is keeping this from being merged besides the conflicts? I'm happy to go through my company's OSS licensing process and help with this ticket, but I need to know the status before I commit to that overhead. :)

**Comment by rongrong** [comment]
Some thoughts at a level: In many situations than we all like, Hive and Presto behaves differently. If we silently process a Hive view as a Presto view, the results could be unexpected to user. So the very least we have to offer is to have session properties to enable/disable it, and it should be disabled by default. It also needs to be explicit that this functionality is actually parsing a Hive view as a Presto view (so it's actually not a Hive view when we run it).

**Comment by amoghmargoor** [comment]
@billywatson The reason @rongrong gave is why we I stopped with this PR. Semantics of Hive views can be different and can have unexpected results on Presto. i can try to rebase it with config option if that is going to help, but I doubt it would be accepted and merged.

**Pull request closed without merging**